src/stores/handlers.store.ts

```diff
@@ -69,109 +121,128 @@ export const useHandlersStore = defineStore('handlers', {
         this.getLightsHandlers.bind(this)
       );
 
-      this.loading = false;
+      this.gettingAudio = this.gettingScreens = this.gettingLights = false;
     },
+    /**
+     * Set the audio handler
+     * @param id - id of the audio
+     * @param newHandler - new handler
+     */
     async setAudioHandler(id: number | number[], newHandler: string | null = null): Promise<void> {
-      try {
-        this.loading = true;
-
-        if (Array.isArray(id)) {
-          await Promise.all(
-            id.map((i) =>
-              HandlersService.setAudioHandler(i, {
-                name: newHandler != null ? newHandler : ''
-              })
-            )
-          );
-        } else {
-          await HandlersService.setAudioHandler(id, {
-            name: newHandler != null ? newHandler : ''
-          });
-        }
-
-        this.audioHandlers = await HandlersService.getAudioHandlers();
-      } catch (e: any) {
-        handleError(e);
+      this.settingAudio = true;
+      if (Array.isArray(id)) {
+        await Promise.all(
+          id.map((i) =>
+            setAudioHandler({
+              body: { name: newHandler != null ? newHandler : '' },
+              path: { id: i }
+            }).catch(noop)
```

Comment:
Any calls make by fetch (and thus all calls made by the API client) are first intercepted and checked for errors. These are displayed using Toast. The `catch` clause is solely left for dealing with possible side-effects of the failed request.

In the original code this was done explicitly. However, this is now dealt with implicitly. No other side-effects were dealt with, so I replaced it with a `noop` instead.

Reply:
Okay, makes sense. However, it seems many/most of these `.catch(noop)` statements are present in asynchronous functions. Just like in other comment, to me it would be nicer to rewrite this (and similar functions) with `await`:

```ts
await setAudioHandler({
  body: { name: newHandler != null ? newHandler : '' },
  path: { id: i }
});
```

By using await in an asynchronous function, you don't need to explicitly define a `.catch()` block that does not do anything.

Reply:
I have removed catch blocks where `noop` operation was used now. In a future PR I will more thorougly check all modules to check what breaks and needsd fallback options when certain backend modules are unavailable for one reason or another.